membership_extension/models/membership_line.py

```diff
@@ -0,0 +1,79 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 Antonio Espinosa <antonio.espinosa@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from datetime import timedelta
+from openerp import models, fields, api, _
+from openerp.exceptions import Warning as UserError
+from openerp.addons.membership.membership import STATE
+
+
+class MembershipLine(models.Model):
+    _inherit = "membership.membership_line"
+    _order = 'date_to desc, id desc'
+
+    category_id = fields.Many2one(
+        string="Category", comodel_name='membership.membership_category',
+        related="membership_id.membership_category_id", readonly=True)
+    date_from = fields.Date(readonly=False)
+    date_to = fields.Date(readonly=False)
+    state = fields.Selection(
+        selection=STATE, string="Membership Status", readonly=False,
+        compute='_compute_state', inverse="_inverse_state",
+        help="""It indicates the membership status.
+        - Non Member: A member who has not applied for any membership.
+        - Cancelled Member: A member who has cancelled his membership.
+        - Old Member: A member whose membership date has expired.
+        - Waiting Member: A member who has applied for the membership and
+                          whose invoice is going to be created.
+        - Invoiced Member: A member whose invoice has been created.
+        - Paid Member: A member who has paid the membership amount."""
+    )
+
+    def __init__(self, pool, cr):
```

Comment:
I have already answered to that: you have to call it manually in your test, like in https://github.com/OCA/server-tools/blob/8.0/base_suspend_security/tests/test_base_suspend_security.py#L28.

It's the only drawback. The rest is better handled this way.